recipes/debian.rb

```diff
@@ -21,19 +21,15 @@
 
 TIMEZONE_FILE = '/etc/timezone'.freeze
 
-bash 'check-localtime' do
-  user 'root'
-  cwd '/tmp'
-  only_if { File.exist?('/etc/localtime') }
-  code <<-EOH
-  ls -la /etc/localtime >> /tmp/check.txt
-  if grep #{node['timezone_iii']['timezone']} /tmp/check.txt ; then
-    echo "Nothing to do!!!"
-  else
-    rm /etc/localtime
-  fi
-  rm /tmp/check.txt
-  EOH
+file 'remove-wrong-localtime-link' do
+  path '/etc/localtime'
+  action :delete
+  manage_symlink_source false
+  only_if {
+    File.exist?('/etc/localtime') &&
+      !File.symlink?('/etc/localtime') ||
```

Comment:
Wouldn't this fail to remove the file if it is a symlink but not the correct symlink?

Reply:
I think it would.
```
!File.symlink?('/etc/localtime') || !File.readlink('/etc/localtime').include?(node['timezone_iii']['timezone'])
```
should be simply read as
```
not symlink or incorrect symlink
```

Reply:
Or better would be file exists or symlink and file includes time zone attribute. Since the file could symlink or hard file.

Reply:
I first thought of that idea, but found it impossible.
`/etc/localtime` does not always include time zone name...

```console
vagrant@default-ubuntu-1604:~$ ls -l /etc/localtime
lrwxrwxrwx 1 root root 35 Dec 11 05:31 /etc/localtime -> /usr/share/zoneinfo/Europe/Istanbul
vagrant@default-ubuntu-1604:~$ cat /etc/localtime | strings
TZif2
EEST
TZif2
EEST
<+03>-3
```

Reply:
@Stromweld any better idea? At least this patch I believe will not change the current released behaviour.

Reply:
I agree this patch doesn't change current behavior. Thinking about it more this removes the file if it is a file and not sym_link or sym_linked to wrong tz file. But then we never set it correctly anywhere. I also don't think the file should necessarily be removed if it's not a sym_link. Personally I prefer it to be sym_link on my systems but some people may have it a hard file that they copied from tz folder and renamed to localtime.

Reply:
I see and I think `/etc/localtime` should be symlink too.

Anyway I'll change the patch not to remove an ordinary (non-symlink) file.
People who want the file to be an ordinary file would write an original recipe that makes a hard copy.

Reply:
I wonder if a better check would be to simply compare the contents to of /etc/localtime to contents of "/usr/share/zoneinfo/#{node['timezone_iii']['timezone']}" That way if it is a hardfile we'll determine if it's correct and if it is a symbolic link to the correct file they'll match. If they don't match then we correct it by removing file and creating sym_link to correct file.

Reply:
That logic seems very simple. I updated the code.